peg_src/jsdoctype.pegjs

```diff
@@ -980,14 +981,17 @@ RecordTypeExprEntries = first:RecordTypeExprEntry restWithComma:(_ "," _ RecordT
                         }, [first]);
                       }
 
-
-RecordTypeExprEntry = keyInfo:RecordTypeExprEntryKey _ ":" _ value:RecordTypeExprEntryOperand {
+RecordTypeExprEntry = keyInfo:RecordTypeExprEntryKey _ optional:"?"? _ ":" _ value:RecordTypeExprEntryOperand {
```

Comment:
I believe that whitespace is disallowed between the property name and `?`.

Reply:
I've tested, TypeScript actually does allow whitespace before the `?`